reactioncommerce/reaction
# (feat): 4246 nnnnat new mocks factory

Resolves #4246   
Impact: **minor**  
Type: **feature|refactor**

## Issue
Writing GQL integration test has become difficult because we've been creating mock data ad-hoc. 
More info [here](https://github.com/reactioncommerce/reaction/issues/4246)

## Solution
 1. Create a new `@reactioncommerce` npm package that can take the de-meteorized `SimpleSchema` and create a mocks factory for each schema. See [data-factory](https://github.com/reactioncommerce/data-factory)
 2. Create a new `test-utils/helper` that will build a mocks factory using the [data-factory](https://github.com/reactioncommerce/data-factory) package and export a `Factory` object to be used in the integration test.
3. Removing some leftover meteor deps from the schemas that we missed in PR [4266](https://github.com/reactioncommerce/reaction/pull/4266).
4. Replacing Tags integration test mock data with mocks from the test utils `Factory`.

## Breaking changes
1. Possible breaking change: RC's product variant schema still had a meteor dep inside one of the properties. This is being removed/refactored to not use the `ReactionProduct` API from `lib/api`

## Testing
1. Spin up a fresh RC shop confirm it still works.
2. Create, edit and archive a new Product
2. Run integration test `docker-compose run devserver npm run test:integration`
